src/gossip_agent/waku_handling.rs

```diff
@@ -269,20 +270,29 @@ pub async fn handle_signal(
                         graphcast_message
                     );
 
-                    let block: Block<_> = provider
-                        .get_block(graphcast_message.block_number)
-                        .await
-                        .unwrap()
-                        .unwrap();
-                    let block_hash = format!("{:#x}", block.hash.unwrap());
+                    if content_topics.iter().any(|content_topic| {
+                        content_topic.content_topic_name == graphcast_message.identifier
```

Comment:
any interest to use the existing helper function `match_content_topic`?

Reply:
done ✅